### ISSUE TITLE
RuntimeError: can't start new thread 

### ISSUE THREAD

**Issue by wilsonmr** [tqdm/tqdm#522]
Apologies in advanced if I haven't filed this correctly, I'm relatively new to this.

I get a `RuntimeError: can't start new thread ` whilst installing packages with conda, and the issue appears to be with when tqdm is called, I filed the issue here https://github.com/conda/conda/issues/7040 and the owner suggested that I should check if it had been reported with upstream tqdm. As far as I can tell this issue hasn't been logged before, nor is it a known issue. 

I have provided an example of how I can get the error, whilst using conda on a login node on a HPC whilst trying to install python to an empty environment in miniconda, I mentioned in https://github.com/conda/conda/issues/7040 that this is possibly caused by limited system-wide resources on the login nodes, and pasted the output of `ulimit -a` if relevant.

If I instead did `conda install python --quiet` then I don't get the issue, this is reproducable for other packages and also using both system python and python 3.6.4 downloaded through conda in quiet mode.

In this example:
Python version: Python 2.7.5
OS: Scientific Linux 7.2 (Nitrogen)
tqdm: 4.14.0 according to https://github.com/conda/conda/blob/master/conda/_vendor/tqdm/__init__.py#L3

```
[s1758208@login04(eddie) ~]$ conda create -n test
Solving environment: done

## Package Plan ##

  environment location: /exports/eddie/scratch/s1758208/minitest/envs/test


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate test
#
# To deactivate an active environment, use
#
#     $ conda deactivate

[s1758208@login04(eddie) ~]$ conda activate test
(test) [s1758208@login04(eddie) ~]$ conda install python
Solving environment: done

## Package Plan ##

  environment location: /exports/eddie/scratch/s1758208/minitest/envs/test

  added / updated specs: 
    - python


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    setuptools-38.5.1          |           py36_0         525 KB
    libstdcxx-ng-7.2.0         |       hdf63c60_3         2.5 MB
    libgcc-ng-7.2.0            |       hdf63c60_3         6.1 MB
    pip-9.0.1                  |           py36_5         2.2 MB
    python-3.6.4               |       hc3d631a_3        29.1 MB
    ------------------------------------------------------------
                                           Total:        40.4 MB

The following NEW packages will be INSTALLED:

    ca-certificates: 2017.08.26-h1d4fec5_0
    certifi:         2018.1.18-py36_0     
    libedit:         3.1-heed3624_0       
    libffi:          3.2.1-hd88cf55_4     
    libgcc-ng:       7.2.0-hdf63c60_3     
    libstdcxx-ng:    7.2.0-hdf63c60_3     
    ncurses:         6.0-h9df7e31_2       
    openssl:         1.0.2n-hb7f436b_0    
    pip:             9.0.1-py36_5         
    python:          3.6.4-hc3d631a_3     
    readline:        7.0-ha6073c6_4       
    setuptools:      38.5.1-py36_0        
    sqlite:          3.22.0-h1bed415_0    
    tk:              8.6.7-hc745277_3     
    wheel:           0.30.0-py36hfd4bba0_1
    xz:              5.2.3-h55aa19d_2     
    zlib:            1.2.11-ha838bed_2    

Proceed ([y]/n)? y


Downloading and Extracting Packages
setuptools 38.5.1: ##################################################### | 100% 
libstdcxx-ng 7.2.0: #################################################### | 100% 

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/exports/eddie/scratch/s1758208/minitest/lib/python3.6/site-packages/conda/exceptions.py", line 789, in __call__
        return func(*args, **kwargs)
      File "/exports/eddie/scratch/s1758208/minitest/lib/python3.6/site-packages/conda/cli/main.py", line 78, in _main
        exit_code = do_call(args, p)
      File "/exports/eddie/scratch/s1758208/minitest/lib/python3.6/site-packages/conda/cli/conda_argparse.py", line 77, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "/exports/eddie/scratch/s1758208/minitest/lib/python3.6/site-packages/conda/cli/main_install.py", line 11, in execute
        install(args, parser, 'install')
      File "/exports/eddie/scratch/s1758208/minitest/lib/python3.6/site-packages/conda/cli/install.py", line 255, in install
        handle_txn(progressive_fetch_extract, unlink_link_transaction, prefix, args, newenv)
      File "/exports/eddie/scratch/s1758208/minitest/lib/python3.6/site-packages/conda/cli/install.py", line 281, in handle_txn
        progressive_fetch_extract.execute()
      File "/exports/eddie/scratch/s1758208/minitest/lib/python3.6/site-packages/conda/core/package_cache.py", line 584, in execute
        exc = self._execute_actions(prec_or_spec, prec_actions)
      File "/exports/eddie/scratch/s1758208/minitest/lib/python3.6/site-packages/conda/core/package_cache.py", line 599, in _execute_actions
        progress_bar = ProgressBar(desc, not context.verbosity and not context.quiet, context.json)
      File "/exports/eddie/scratch/s1758208/minitest/lib/python3.6/site-packages/conda/common/io.py", line 390, in __init__
        self.pbar = tqdm(desc=description, bar_format=bar_format, ascii=True, total=1)
      File "/exports/eddie/scratch/s1758208/minitest/lib/python3.6/site-packages/conda/_vendor/tqdm/_tqdm.py", line 388, in __new__
        cls.monitor = TMonitor(cls, cls.monitor_interval)
      File "/exports/eddie/scratch/s1758208/minitest/lib/python3.6/site-packages/conda/_vendor/tqdm/_tqdm.py", line 83, in __init__
        self.start()
      File "/exports/eddie/scratch/s1758208/minitest/lib/python3.6/threading.py", line 846, in start
        _start_new_thread(self._bootstrap, ())
    RuntimeError: can't start new thread
```

**Comment by kalefranz** [comment]
Hello. Lead dev of conda here.  This issue was reported on our issue tracker.  I think the stack trace makes pretty clear what's going on.  We had a similar thread issue with concurrent.futures, and the solution that appears to have worked is https://github.com/conda/conda/pull/6653.

We're vendoring tqdm into conda at `conda/_vendor/tqdm`. Our current version is 4.14.0 vendored on 2017-06-18.

https://github.com/conda/conda/blob/deb16ee5dbfa68ae3433f1cc2766999613914b20/conda/_vendor/tqdm/__init__.py#L2-L4

It's very possible that this issue has already been resolved, and we just need to update.  However I didn't see any mention of such RuntimeError so far in your issue tracker.

**Comment by casperdcl** [comment]
Ah, so conda uses tqdm now. Thanks, I think the issue os `tqdm`'s beta support for multithreading. Your issue would probably be resolved by (right after importing tqdm):

```python
from tqdm import tqdm
tqdm.monitor_interval = 0
```

as mentioned in the documentation. Basically we start a monitoring thread to make sure the progress bar always updates at least every 10 seconds, but it seems to be causing you issues.

**Comment by casperdcl** [comment]
Really, we could catch and ignore the error during the monitor creation...

**Comment by kalefranz** [comment]
Cool.  If you think you might have a patch soon that ignores the error, I'll wait and just update the vendored tqdm with the new version.  Otherwise we'll set `monitor_level` as you suggest.

**Comment by kalefranz** [comment]
I do prefer the patch that ignores the error though.  There's certainly value to the monitor for most users, but it's also more optional.  My preference would be to keep it enabled.

**Comment by casperdcl** [comment]
yes, just pushed a patch to `devel`. Probably don't want an warning message either? What do you think?

`msg = "tqdm:disabling monitor support (monitor_interval = 0)\n"`

1. `warnings.warn(msg)`
1. `sys.stderr.write(msg)`
1. `print(msg)`
1. `raise TqdmWarning(msg)`

Personally I'd prefer the first, then you could choose to filter out the warning in conda if you want

**Comment by kalefranz** [comment]
Do you use the python logger at all?  If so, my preference would be `log.info()`.

**Comment by kalefranz** [comment]
Actually, I don't really care.  I don't think many users hit this.  And warnings should always go to stderr.  So my preference order would be

1. `log.info(msg)`
2. `warnings.warn(msg)`
3. `print(msg, file=sys.stderr)`

**Comment by casperdcl** [comment]
curious - you don't suggest `log.warn`?

**Comment by kalefranz** [comment]
Thanks for the *very* quick turnaround here! Do you have a ballpark estimate on when this will be tagged into a release? Does the v5.0.0 milestone imply that you're targeting the next release off of master to be 5.0.0?

**Comment by kalefranz** [comment]
Nevermind.  I see that you released 4.19.8. 👍👍

**Comment by casperdcl** [comment]
Well I try to be quick with critical (ie raise errors rather than warnings) bugs that are easy to fix.
